src/components/MessageView.tsx

```diff
@@ -42,6 +46,41 @@ export default function MessageView (props: { message: Message; id: string }) {
   return React.createElement(component, { ...message, id: props.id })
 }
 
+const handleDeleteMessage = (e, data) => {
+  const doDelete = confirm(`Are you sure you would like to delete the message '${data.message}'?`)
+  if (doDelete) {
+    deleteMessage(data.messageId)
+  }
+}
+
+const DeleteMessageView = (props: { messageId: string, message: string, id: string }) => {
+  const { userMap, myId } = useContext(UserMapContext)
+  const playerIsMod = userMap[myId] && userMap[myId].isMod
+
+  if (!playerIsMod) {
+    return (
+      <span>{props.message}</span>
+    )
+  } else {
+    return (
+      <span className="deleteMenu">
+        <ContextMenuTrigger id={props.messageId} renderTag="span" holdToDisplay={0}>
+          {props.message}
```

Comment:
Instead of passing in `props.message` as a property, it might make sense to use `props.children` here. That would let us do something like e.g.

`<DeleteMessageView id='1234' messageId='5678'>I am a <strong>message with nested tags!</strong></DeleteMessageView>` and have the nested React components properly rendered.

Feels a bit more React-y, and might be helpful with new message types? 

(https://reactgo.com/react-props-children/ if you haven't seen this yet)